sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -372,6 +378,51 @@ public EventHubClientBuilder prefetchCount(int prefetchCount) {
         return this;
     }
 
+    /**
+     * Enables idempotent publishing when an {@link EventHubProducerAsyncClient} or {@link EventHubProducerClient}
+     * is built.
+     *
+     * If enabled, the producer will only be able to publish directly to partitions; it will not be able to publish to
+     * the Event Hubs gateway for automatic partition routing nor using a partition key.
+     *
+     * @return The updated {@link EventHubClientBuilder} object.
+     */
+    public EventHubClientBuilder idempotentPartitionPublishing() {
```

Comment:
```suggestion
    public EventHubClientBuilder enableIdempotentPartitionPublishing() {
```

Reply:
Changed to enableIdempotentPartitionPublishing for this preview. Let's discuss later what's the best way for boolean value in builders.